### ISSUE TITLE
Syncing large Email account does not work well.

### ISSUE THREAD

**Issue by Iain-Stevenson** [nylas/N1#1004]
I have an outlook email account with Many nested folders and approx 3000+ emails. These emails are a vital audit of business transactions and have to be kept for years. I want to dump microsoft outlook in favour of N1 but cannot as it stands because there are fundamental shortcomings;

The sync operation for this account is problematic. It gives up before its done the whole task.

You cannot confirm with any known message counts per account or folder in N1

I also cannot find a way to change the account details, e.g. password in the account management views.

The sync seemed to work really quickly but then I realized that not all of the folders have their messages. 

I performed several hopeful searches to try and force a sync as there is no manual sync button. I think that a manual sync instigation button is needed on the button / task bar

In order to confirm whether this works OK or not I now have to manually traverse the entire folder structure and eyeball and the count the number of items as there is no folder item count available. This is needed as default. This is an impossibly long task.

Please add folder counts along side the unread count i.e. 2 unread of 124  [2/124] as this would help.plus a total for the account.

**Comment by Iain-Stevenson** [comment]
Update: The syncing seems to be working in background, and more folders are filling over time. 

there is 
1) no visual information to say that a sync is still ongoing.
2) no means by which any progress can be determined at all, or when the syncing is done.

**Comment by Iain-Stevenson** [comment]
Still love the UI though, when this product matures a little more it will be very good indeed.

**Comment by jstejada** [comment]
Hi @Iain-Stevenson , just to chime in regarding unread counts for every folder, you can enable them in `Preferences -> General` and selecting `Show unread counts for all folders/labels`

**Comment by Iain-Stevenson** [comment]
Yeah I spotted that but I mean it needs the total counts as well.

I.e. # unread of / # total

  

e.g. 2 / 124

  

But thanks for the heads up. :-)

  

TBH the in the context of the entire sync we really need a percent complete
i.e. progress bar style visual.

  

the 'syncing account' display does appear at the bottom of the left hand pane
after certain actions but is only there for seconds on what must be a long
running sync. I'm not sure why its saying its finished  

  

 I'm not sure if its outlook.com that's the problem or not. Could well be

  

I had trouble getting that account registered and was forced to use the IMAP
instead of exchange (Outlook) option.

  

My other account went in as and Outlook/exchange account ok but it too took
much shoving to sync completely and again I have no idea if its finished as it
too has thousands of messages in archive.

  

  

BTW I log the way this is tablet friendly!!!!

  

  

> On Jan 13 2016, at 10:54 pm, Juan &lt;notifications@github.com&gt; wrote:  

>

> Hi [@Iain-Stevenson](https://github.com/Iain-Stevenson) , just to chime in
regarding unread counts for every folder, you can enable them in Preferences
-&gt; General and selecting Show unread counts for all folders/labels

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/nylas/N
1/issues/1004#issuecomment-171463146).![](https://github.com/notifications/bea
con/ABkQ7x346UzKMwAG9O2BZvny_Hdd3Hx3ks5pZs0KgaJpZM4HEZDS.gif)

**Comment by Iain-Stevenson** [comment]
Having got a pointer to some on line docs at Nylas I now think I realize what is happening.

The docs suggest that the sync is in fact continuing at the Nylas server and updating the UI on demand. It also may be influenced by throttling at the end mail host service. Which is why I see changes in the folder contents from time to time without seeing a 'sync in progress' visual.  The syncing progress indicator only appears briefly after perform actions like searching.

My confidence is much higher that all of the emails are being 'obtained' from outlook.com but it was a bit worrying there for a while.

I'd still like to see a unread/total badge by each folder though as this gives me lots of quick info for checking AND organising and potentially cleaning folders.

**Comment by jackiehluo** [comment]
Another incidence: https://twitter.com/willdayble/status/747209342068264960

**Comment by jackiehluo** [comment]
Another incidence: https://nylas.zendesk.com/agent/tickets/4074

**Comment by jackiehluo** [comment]
Another incidence: https://nylas.zendesk.com/agent/tickets/4414

**Comment by jackiehluo** [comment]
Another incidence: https://nylas.zendesk.com/agent/tickets/4433
